### PR TITLE
Addressed bug fixes from 1.3.0

### PR DESCRIPTION
**Pull request by jeffmay**
@egprentice @audax-shreyaspurohit

### CODE REVIEW

**Comment by egprentice** [comment]
LGTM

**Comment by audax-shreyaspurohit** [comment]
lgtm

**Comment by jeffmay** [comment]
~~FAIL: Intermittent test failiures~~

Okay, I fixed some bugs with unique test name generation, but I couldn't figure out why they won't work when run in an async fashion. I am tabling this for now. I'll look into it down the road.

File: bson-adt-core/src/main/scala/adt/bson/Bson.scala [hunk 1]
```diff
@@ -42,8 +47,253 @@ object Bson {
...
+   */
+  def pretty(bson: BsonValue): String = stringify(bson)(StringifyFormat.pretty)
+
+  /**
+   * Prints the bson in the least amount of characters on a single line.
+   */
+  def inline(bson: BsonValue): String = stringify(bson)(StringifyFormat.inline)
+
+  /**
+   * Prints the bson using the custom style profived implicitly.
```
Comment (egprentice) [inline bson-adt-core/src/main/scala/adt/bson/Bson.scala]: profived -> provided

File: bson-adt-core/src/test/java/org/google/plaintext/diff_match_patch_test.java [hunk 2]
```diff
@@ -0,0 +1,908 @@
...
+    //  DIFF TEST FUNCTIONS
+
+
+    public void testDiffCommonPrefix() {
+        // Detect any common prefix.
+        assertEquals("diff_commonPrefix: Null case.", 0, dmp.diff_commonPrefix("abc", "xyz"));
+
+        assertEquals("diff_commonPrefix: Non-null case.", 4, dmp.diff_commonPrefix("1234abcdef", "1234xyz"));
+
+        assertEquals("diff_commonPrefix: Whole case.", 4, dmp.diff_commonPrefix("1234", "1234xyz"));
```
Comment (egprentice) [inline bson-adt-core/src/test/java/org/google/plaintext/diff_match_patch_test.java:54]: also try
 assertEquals("diff_commonPrefix: Whole case.", 4, dmp.diff_commonPrefix("1234xyz", "1234"));

File: bson-adt-core/src/test/java/org/google/plaintext/diff_match_patch_test.java [hunk 3]
```diff
@@ -0,0 +1,908 @@
...
+    }
+
+    public void testDiffCommonSuffix() {
+        // Detect any common suffix.
+        assertEquals("diff_commonSuffix: Null case.", 0, dmp.diff_commonSuffix("abc", "xyz"));
+
+        assertEquals("diff_commonSuffix: Non-null case.", 4, dmp.diff_commonSuffix("abcdef1234", "xyz1234"));
+
+        assertEquals("diff_commonSuffix: Whole case.", 4, dmp.diff_commonSuffix("1234", "xyz1234"));
```
Comment (egprentice) [inline bson-adt-core/src/test/java/org/google/plaintext/diff_match_patch_test.java:63]: also try
assertEquals("diff_commonSuffix: Whole case.", 4, dmp.diff_commonSuffix("xyz1234", "1234"));

File: bson-adt-core/src/test/java/org/google/plaintext/diff_match_patch_test.java [hunk 4]
```diff
@@ -0,0 +1,908 @@
...
+
+    public void testDiffCommonOverlap() {
+        // Detect any suffix/prefix overlap.
+        assertEquals("diff_commonOverlap: Null case.", 0, dmp.diff_commonOverlap("", "abcd"));
+
+        assertEquals("diff_commonOverlap: Whole case.", 3, dmp.diff_commonOverlap("abc", "abcd"));
+
+        assertEquals("diff_commonOverlap: No overlap.", 0, dmp.diff_commonOverlap("123456", "abcd"));
+
+        assertEquals("diff_commonOverlap: Overlap.", 3, dmp.diff_commonOverlap("123456xxx", "xxxabcd"));
```
Comment (egprentice) [inline bson-adt-core/src/test/java/org/google/plaintext/diff_match_patch_test.java:74]: also try
assertEquals("diff_commonOverlap: Overlap.", 3, dmp.diff_commonOverlap("xxx123456", "abcdxxx"));

**Comment by egprentice** [comment]
LGTM, (some minor comments in the jUnit test cases)

**Pull request merged**